internal/iam/repository_group.go

```diff
@@ -90,33 +92,34 @@ func (r *Repository) UpdateGroup(ctx context.Context, group *Group, version uint
 			g := group.Clone().(*Group)
 			resource, rowsUpdated, err = r.update(ctx, g, version, dbMask, nullFields)
 			if err != nil {
-				return err
+				return errors.Wrap(err, op)
 			}
 			repo, err := NewRepository(read, w, r.kms)
 			if err != nil {
-				return fmt.Errorf("update group: failed creating inner repo: %w for %s", err, group.PublicId)
+				return errors.Wrap(err, op, errors.WithMsg(fmt.Sprintf("for group %s", group.PublicId)))
```

Comment:
you've lost traceability between this error and the next one.. it might be ambiguous which error happen

Reply:
Yeah so the issue here really is around having the redundant op from the wrapped error. as I discuss in a different thread.  So in this example it's between:
```iam.(Repository).UpdateGroup: failed creating inner repo: iam.NewRepository: org error```
and
```iam.(Repository).UpdateGroup: iam.NewRepository: org error```
I felt the iam.NewRepository already clarifies the msg `failed creating inner repo`.  I do however agree this does make a simple search for a specific string not as easy